watertap/unit_models/tests/test_translator_adm1_asm1.py

```diff
@@ -0,0 +1,303 @@
+        m.fs.unit.inlet.conc_mass_comp[0, "X_ac"].fix(677.24 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_h2"].fix(284.84 * units.mg / units.liter)
+        m.fs.unit.inlet.conc_mass_comp[0, "X_I"].fix(17216 * units.mg / units.liter)
+
+        m.fs.unit.inlet.cations[0].fix(1e-8 * units.mmol / units.liter)
+        m.fs.unit.inlet.anions[0].fix(5.21 * units.mmol / units.liter)
+
+        return m
+
+    @pytest.mark.build
+    @pytest.mark.unit
+    def test_build(self, asmadm):
+
+        assert hasattr(asmadm.fs.unit, "inlet")
+        assert len(asmadm.fs.unit.inlet.vars) == 6
+        assert hasattr(asmadm.fs.unit.inlet, "flow_vol")
+        assert hasattr(asmadm.fs.unit.inlet, "conc_mass_comp")
+        assert hasattr(asmadm.fs.unit.inlet, "temperature")
+        assert hasattr(asmadm.fs.unit.inlet, "pressure")
+        assert hasattr(asmadm.fs.unit.inlet, "anions")
+        assert hasattr(asmadm.fs.unit.inlet, "cations")
+
+        assert hasattr(asmadm.fs.unit, "outlet")
+        assert len(asmadm.fs.unit.outlet.vars) == 5
+        assert hasattr(asmadm.fs.unit.outlet, "flow_vol")
+        assert hasattr(asmadm.fs.unit.outlet, "conc_mass_comp")
+        assert hasattr(asmadm.fs.unit.outlet, "temperature")
+        assert hasattr(asmadm.fs.unit.outlet, "pressure")
+        assert hasattr(asmadm.fs.unit.outlet, "alkalinity")
+
+        assert number_variables(asmadm) == 129
+        assert number_total_constraints(asmadm) == 16
+
+        assert number_unused_variables(asmadm.fs.unit) == 4
+
+    @pytest.mark.component
+    def test_units(self, asmadm):
+        assert_units_consistent(asmadm)
+
+    @pytest.mark.unit
+    def test_dof(self, asmadm):
+        assert degrees_of_freedom(asmadm) == 0
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_initialize(self, asmadm):
+        initialization_tester(asmadm, optarg={"bound_push": 1e-8})
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, asmadm):
+        solver = get_solver(options={"bound_push": 1e-8})
+        results = solver.solve(asmadm)
+        assert_optimal_termination(results)
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solution(self, asmadm):
+        assert pytest.approx(101325.0, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.pressure[0]
+        )
+        assert pytest.approx(308.15, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.temperature[0]
+        )
+        assert pytest.approx(0.1308, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "S_I"]
+        )
+        assert pytest.approx(0.2585, abs=1e-2) == value(
+            asmadm.fs.unit.outlet.conc_mass_comp[0, "S_S"]
```

Comment:
We typically used relative tolerances and reserve absolute tolerances for checking 0 values. Additionally, preferable to check relative tolerance of 1e-3 to 1e-5, if we're being consistent with other testing on WaterTAP.